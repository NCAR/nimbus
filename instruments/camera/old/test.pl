#!/usr/bin/perl
$i = 0;
my @array = (1,2,3,4,5,6);
splice @array, 2, 1;
splice @array, $i, 1;
print join(":", @array);
print "\n";
