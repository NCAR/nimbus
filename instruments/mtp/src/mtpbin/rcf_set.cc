#include <stdio.h>
#include <string.h> 
#include <iostream>
#include <fstream>
#include <arpa/inet.h>
#include <boost/filesystem.hpp>
#include <math.h>
#include "rcf_set.h"
#include "rcf.h"

using namespace std;

RetrievalCoefficientFileSet::RetrievalCoefficientFileSet(const string Directory) 
{

  _RCFDir = Directory;
  boost::filesystem::path RCFPath(_RCFDir);

  boost::filesystem::directory_iterator end_itr;

  for (boost::filesystem::directory_iterator itr(RCFPath); itr != end_itr; 
       ++itr)
  {
    // Ignore directories and files that don't have .RCF extension
    int i = 0;
    if (boost::filesystem::is_regular_file(itr->path()) && 
        itr->path().extension() == ".RCF") {
      string rcf_path = itr->path().string();
      _RCFs.push_back(RetrievalCoefficientFile(rcf_path));
      //cout << "Found RCF file:"<<rcf_path<<"  with ID:"
       //    <<_RCFs[i].getId()<<"\n";
      i++;
    }
  }

  return;
}

bool RetrievalCoefficientFileSet::setFlightLevelsKm(float FlightLevels[], 
                                                    int NumFlightLevels)
{
  if (_RCFs.size() == 0) 
  {
    cout<<"In: "<<__func__<<" call failed:\n";
    cout<<"ERROR:There are currently no RCFs in the set\n";
    return false;
  } 

  for (std::vector<RetrievalCoefficientFile>::iterator 
       it = _RCFs.begin(); it != _RCFs.end(); ++it) 
  {
    if (!it->setFlightLevelsKm(FlightLevels, NumFlightLevels))
    {
      cout<<"In: "<<__func__<<" call failed:\n";
      cout<<"ERROR: Failed to set flight levels for RCFID:"<<it->getId()<<"\n";
      return false;
    }
  }

  return true;
}

RC_Set_1FL RetrievalCoefficientFileSet::getBestWeightedRCSet
                                  (float ScanBrightnessTemps[],float PAltKm,
                                   float BTBias)
{

  RC_Set_1FL AvgWtSet;
  float Weight, SumWeights;         // For RMS weighting
  float SumWeightedAvg, SumSquares; // Sum of weighted differences
  float Diff;			    // Measured BT - Model BT - BTBias
  int   NumBTsIncl;		    // Count of BTs included in Weighting
  int   BestRCIndex;                // Index of "best" template (so far)
  int   BTIndex;		    // Index into Scan and Model BT arrays
  int   thisRCFIndex=0;		    // Which RCF index are we looking at?
  float RCFBTWeightedMean;	    // The weighted mean of this RCF's BTs
  float RCFBTStdDev;		    // Standard Deviation about weighted mean(?)
  float Numerator, Denominator;	    // For RCFBTStdDev calculation
  float BestlnP;  // The sum of the ln of Probabilities for the BestRCIndex
  float thislnP;  // The sum of the ln of Probabilities for "this" RCF


  // Step through the vector of Retrieval Coefficient files (aka templates)
  // to obtain the best match for the scan at the input Altitude
  for (std::vector<RetrievalCoefficientFile>::iterator
       RCFit = _RCFs.begin(); RCFit != _RCFs.end(); ++RCFit)
  {
    SumWeightedAvg = SumSquares = SumWeights = NumBTsIncl = 0;

    // Get the weighted RC set for this template
    AvgWtSet=RCFit->getRCAvgWt(PAltKm);

    // Compare the template Brightness Temperatures with the 
    // measured brightness temperatures to find the quality of match
    for (BTIndex = 0; BTIndex < NUM_BRT_TEMPS; BTIndex++) 
    {
      Weight = 1/(AvgWtSet.MBTRms[BTIndex]*AvgWtSet.MBTRms[BTIndex]);
      SumWeights = SumWeights + Weight;
      Diff = ScanBrightnessTemps[BTIndex]-AvgWtSet.MBTAvg[BTIndex];
      if (Weight > 0)
      {
        NumBTsIncl+=1;
        SumWeightedAvg = SumWeightedAvg+Weight*Diff;
        SumSquares = SumSquares+Weight*(Diff*Diff);
      }
    }

    RCFBTWeightedMean = SumWeightedAvg/SumWeights;

    Numerator = (SumSquares-(SumWeights*(RCFBTWeightedMean*RCFBTWeightedMean)));
    Denominator = ((NumBTsIncl-1)*SumWeights/NumBTsIncl);
    if (Numerator/Denominator >= 0)
      RCFBTStdDev = sqrt(Numerator/Denominator);
    else
      RCFBTStdDev = RCFBTWeightedMean; // MJ "kludge" 

    // Calculate the Sum of the ln of probabilities (quality of match)
    thislnP = 8 * sqrt(RCFBTWeightedMean*RCFBTWeightedMean +
                       RCFBTStdDev*RCFBTStdDev) / NUM_BRT_TEMPS;

    if (RCFit == _RCFs.begin())
    {
      BestlnP = thislnP;
      BestRCIndex = 0;
    }
    else if (thislnP < BestlnP)
    {
      BestlnP = thislnP;
      BestRCIndex = thisRCFIndex;
    }
std::cout<<"Index:";
std::cout<<thisRCFIndex<<"  lnP:";
std::cout<<thislnP<<"\n";
    thisRCFIndex++;
  }

  return _RCFs[BestRCIndex].getRCAvgWt(PAltKm);

}

