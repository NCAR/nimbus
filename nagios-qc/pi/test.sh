#! / bin / bash           
output=$(psg radar)
echo "$output"
