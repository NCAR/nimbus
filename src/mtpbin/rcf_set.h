/*
 *************************************************************************** 
 ** rcf_set.h
 ** 
 ** The Microwave Temperature Profiler (MTP) instrument build by JPL for
 ** NCAR as part of the set of instruments known as HAIS (build specifically
 ** in support of the Haiper GV aircraft) came with processing software 
 ** written in Visual Basic 6.0. 
 **
 ** A set of Retrieval Coefficient Files (RCFs) are currently written 
 ** by the VB program RCCalc which takes RAOB data and converts it into 
 ** Templates that describe what that RAOB profile would look like to the MTP 
 ** instrument if the instrument was used to measure the atmosphere described 
 ** by the profile. 
 **
 ** When the MTP instrument completes a scan of the atmosphere the scan counts
 ** are converted to Brightness Temperatures, using these the "best match" 
 ** RCF is found and it's corresponding Retrieval Coefficients are used
 ** to determine the atmospheric temperature profile.
 ** 
 ** This class encapsulates the whole set of RCFs for a project and provides
 ** a method for determining which of the RCFs is the best match of a given
 ** set of brightness temperatures (scan) taken at a particular elevation.
 **
 ** Copyright 2015-2016 University Corporation for Atmospheric Research
 **  - VB6 and Algorithm Copyright MJ Mahoney, NASA Jet Propulsion Laboratory
 **
*/
#ifndef _RCF_SET_h
#define _RCF_SET_h

#include <vector>
#include <string> 
#include "rcf.h"

class RetrievalCoefficientFileSet  
{

public:

  std::vector<RetrievalCoefficientFile> getRCFVector() {return _RCFs;}

  /**
   * Constructor
   * @param std::string is directory name containing the RCF files to be 
   * pulled into the set.
   */
  RetrievalCoefficientFileSet(const std::string);

  /**
   * Set the Flight Levels in KM.  
   * @param[in] is an array of floating point values indicating the km
   * above sea level of each flight level in each retrieval coefficient file.
   * @param[in] is the size fo the vector
   * Flight levels should be ordered in decreasing altitude.
   * @returns true if levels successfully set, false if not.
   * Assumes that all RCFs in the set have the same flight levels - 
   * historically that has been the case, but it may not have to be.
   */
  bool setFlightLevelsKm(float[], int);

  /**
   * Get the Retrieval Coefficient Set from the "best" template
   * weighted according to the flight level of the aircraft and given
   * a set of Scan Brightness Temperatures
   * @param[in] is an array of Scan Brightness Temperatures and is expected
   *  to be NUM_OBSVBLSs in length.
   * @param[in] is the altitude of the aircraft in km at the time that the 
   *  Scan was taken.
   * @pram [in] is an Brightness Temperature Bias provided by user
   * @return is the weighted average set of Model Brightness Temperatures, 
   *  RMS values, and Retrieval Coefficients from the template that best 
   *  matches the input Scan Brightness Temperatures as observed from 
   *  the input flight altitude. 
   */
  RC_Set_1FL getBestWeightedRCSet(float*, float, float);

private: 

  std::string _RCFDir;
  std::vector<RetrievalCoefficientFile> _RCFs;
  
};

#endif
