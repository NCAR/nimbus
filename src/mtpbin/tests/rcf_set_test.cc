#include <iostream>
#include <fstream>
#include <stdlib.h>

#include "../rcf_set.h"
#include "test.h"

using namespace std;

/*
 * This program is designed to test the RetrievalCoefficientFileSet class
 * to assure that it's performing as expected.  The current output file:
 * rcf_set_test.out has been vetted to assure that it has correct information
 * resulting from the tests this program peforms.
 *
 * Usage:
 * rcf_set_test > rcf_set_test.new
 * diff rcf_set_test.out rcf_set_test.new
 *
 * There should be no differences found.  If differences are found then 
 * something has affected the functionality of the class and should be 
 * investigated and resolved.
 */

void rcf_set_test(std::vector<float>,double,std::string,RetrievalCoefficientFileSet);

main () {

  /* Test initialization with a bogus directory */
  RetrievalCoefficientFileSet Bad_RCF_Set(std::string("../"));

  // Test error handling for setting of flight levels
  cout << "Check that setting flight levels on uninitialized set fails. \n";
  Bad_RCF_Set.setFlightLevelsKm(flightLevelsKmH, numFlightLevelsH);

  /* Now initialize with a valid RCF directory */
  RetrievalCoefficientFileSet RCF_Set(RCFdirH);

  std::vector<RetrievalCoefficientFile> RCFs = RCF_Set.getRCFVector();

  RCF_HDR Rcf_Hdr;

  for (std::vector<RetrievalCoefficientFile>::iterator
       it = RCFs.begin(); it != RCFs.end(); ++it)
  {
    cout << "\n\nFound RCF file:"<<it->getFileName()<<" with ID:"<<it->getId()
         <<"\n";
    Rcf_Hdr = it->getRCF_HDR();
    cout << "Raobcount :" << Rcf_Hdr.RAOBcount << "\n";
    cout << "NFL:"<< Rcf_Hdr.NFL<<"\n";
    cout << "First Smatrix2:"<<Rcf_Hdr.SmatrixN2[0][0][0]<<"\n";
  }

  /* Set flight levels */
  RCF_Set.setFlightLevelsKm(flightLevelsKmH, numFlightLevelsH);

  /* Validate the function for selecting an RCF (template) from the set */
  rcf_set_test(scanBTsH1,ACAltKmH1,FlightDescH1,RCF_Set);

  rcf_set_test(scanBTsH3,ACAltKmH3,FlightDescH3,RCF_Set);
}



void rcf_set_test(std::vector<float> scanBTs,double ACAltKm,std::string FlightDesc,
	RetrievalCoefficientFileSet RCF_Set) {

  RC_Set_4Retrieval BestWtdRCSet;
  std::vector<RC_Set_1FL> FlRcSetVec;
  std::vector<int>::size_type sz;
  RC_Set_1FL AvgWtSet;

  // Validate that the function for selecting an RCF (template) from the
  // set is still functioning properly.  Provide a couple of scan values 
  // and verify that the correct RCF is selected.
  cout<<"About to call getBestWeightedRCSet\n";
  BestWtdRCSet = RCF_Set.getBestWeightedRCSet(scanBTs, ACAltKm, 0.0);
  RetrievalCoefficientFile TempRCF(RCF_Set.getRCFbyId(BestWtdRCSet.RCFId));

  // Provide some verification that math going into RCF selection is correct
  // by looking at the avg weighted flight level data.
  if (TempRCF.isValid()) 
  {
    cout<<"BestWeightedRCSet is from file:"
        <<BestWtdRCSet.RCFFileName.c_str()<<"\n";
    FlRcSetVec = TempRCF.getFL_RC_Vec();
    sz = FlRcSetVec.size();
    std::cout <<"PAltKm:"<<ACAltKm<<" 1st level:"<<FlRcSetVec.begin()->Palt
              <<"  last level:"<<FlRcSetVec.end()->Palt<<"  -1:"
              <<FlRcSetVec[sz-2].Palt<<"\n";
    AvgWtSet = TempRCF.getRCAvgWt(ACAltKm);
    std::cout<<"Num,   Avg,  Rms\n";
    for (int i = 0; i < NUM_BRT_TEMPS; i++) 
    {
      std::cout<<i<<", "<<AvgWtSet.MBTAvg[i]<<", "<<AvgWtSet.MBTRms[i]<<"\n";
    }
    cout<<" RCwt:\n";
    for (int j = 0; j < NUM_RETR_LVLS; j++) {
      for (int i = 0; i < NUM_BRT_TEMPS; i++) {
        if (i%5 != 0) cout<<"["<<j<<"]["<<i<<"]:"<<AvgWtSet.RC[j][i];
        else cout<<"["<<j<<"]["<<i<<"]:"<<AvgWtSet.RC[j][i]<<'\n';
      }
    }
    cout <<"\n\n";
  }

  cout<<"Best Retrieval Coef for "<<FlightDesc<<":\n"
      <<"  ID:"<<BestWtdRCSet.RCFId.c_str()<<"\n  "
      <<"Filename:"<<BestWtdRCSet.RCFFileName.c_str()<<"\n  "
      <<"SumLnProb:"<<BestWtdRCSet.SumLnProb<<"\n";
}
