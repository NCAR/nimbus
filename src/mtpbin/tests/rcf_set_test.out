Check that setting flight levels on uninitialized set fails. 
In: setFlightLevelsKm call failed:
ERROR:There are currently no RCFs in the set
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLJ189.RCF with ID:NRCLJ189
Raobcount :190
NFL:15
Last Smatrix2:-0.00166649
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKW189.RCF with ID:NRCKW189
Raobcount :195
NFL:15
Last Smatrix2:-0.00170668
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKZ189.RCF with ID:NRCKZ189
Raobcount :191
NFL:15
Last Smatrix2:-0.00170149
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLS189.RCF with ID:NRCLS189
Raobcount :189
NFL:15
Last Smatrix2:-0.00166569
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKG189.RCF with ID:NRCKG189
Raobcount :186
NFL:15
Last Smatrix2:-0.00167259
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKP189.RCF with ID:NRCKP189
Raobcount :184
NFL:15
Last Smatrix2:-0.00170519
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKO189.RCF with ID:NRCKO189
Raobcount :189
NFL:15
Last Smatrix2:-0.00166861
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLQ189.RCF with ID:NRCLQ189
Raobcount :111
NFL:15
Last Smatrix2:-0.00162657
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKL189.RCF with ID:NRCKL189
Raobcount :127
NFL:15
Last Smatrix2:-0.00165554
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLC189.RCF with ID:NRCLC189
Raobcount :187
NFL:15
Last Smatrix2:-0.0016613
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKX189.RCF with ID:NRCKX189
Raobcount :190
NFL:15
Last Smatrix2:-0.00170021
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLB189.RCF with ID:NRCLB189
Raobcount :188
NFL:15
Last Smatrix2:-0.00169916
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLU189.RCF with ID:NRCLU189
Raobcount :199
NFL:15
Last Smatrix2:-0.00167469
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLE189.RCF with ID:NRCLE189
Raobcount :177
NFL:15
Last Smatrix2:-0.00166323
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLO189.RCF with ID:NRCLO189
Raobcount :190
NFL:15
Last Smatrix2:-0.00170355
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKA189.RCF with ID:NRCKA189
Raobcount :156
NFL:15
Last Smatrix2:-0.0016798
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKS189.RCF with ID:NRCKS189
Raobcount :195
NFL:15
Last Smatrix2:-0.00170317
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKM189.RCF with ID:NRCKM189
Raobcount :198
NFL:15
Last Smatrix2:-0.00169081
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKD189.RCF with ID:NRCKD189
Raobcount :190
NFL:15
Last Smatrix2:-0.00167576
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLA189.RCF with ID:NRCLA189
Raobcount :199
NFL:15
Last Smatrix2:-0.00170223
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLH189.RCF with ID:NRCLH189
Raobcount :170
NFL:15
Last Smatrix2:-0.00168076
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKK189.RCF with ID:NRCKK189
Raobcount :197
NFL:15
Last Smatrix2:-0.00169016
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKH189.RCF with ID:NRCKH189
Raobcount :187
NFL:15
Last Smatrix2:-0.00167013
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKN189.RCF with ID:NRCKN189
Raobcount :114
NFL:15
Last Smatrix2:-0.00166498
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKE189.RCF with ID:NRCKE189
Raobcount :166
NFL:15
Last Smatrix2:-0.00168203
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLD189.RCF with ID:NRCLD189
Raobcount :187
NFL:15
Last Smatrix2:-0.00165552
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKT189.RCF with ID:NRCKT189
Raobcount :194
NFL:15
Last Smatrix2:-0.00170216
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKR189.RCF with ID:NRCKR189
Raobcount :191
NFL:15
Last Smatrix2:-0.00170252
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLK189.RCF with ID:NRCLK189
Raobcount :190
NFL:15
Last Smatrix2:-0.00167774
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLP189.RCF with ID:NRCLP189
Raobcount :114
NFL:15
Last Smatrix2:-0.00164612
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKB189.RCF with ID:NRCKB189
Raobcount :158
NFL:15
Last Smatrix2:-0.001677
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKV189.RCF with ID:NRCKV189
Raobcount :187
NFL:15
Last Smatrix2:-0.00170399
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKU189.RCF with ID:NRCKU189
Raobcount :187
NFL:15
Last Smatrix2:-0.0017037
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLF189.RCF with ID:NRCLF189
Raobcount :144
NFL:15
Last Smatrix2:-0.00165993
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKC189.RCF with ID:NRCKC189
Raobcount :171
NFL:15
Last Smatrix2:-0.00167625
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKJ189.RCF with ID:NRCKJ189
Raobcount :163
NFL:15
Last Smatrix2:-0.00167814
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLI189.RCF with ID:NRCLI189
Raobcount :198
NFL:15
Last Smatrix2:-0.0016921
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLN189.RCF with ID:NRCLN189
Raobcount :189
NFL:15
Last Smatrix2:-0.00170452
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKY189.RCF with ID:NRCKY189
Raobcount :189
NFL:15
Last Smatrix2:-0.00170359
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLT189.RCF with ID:NRCLT189
Raobcount :174
NFL:15
Last Smatrix2:-0.00169708
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLG189.RCF with ID:NRCLG189
Raobcount :197
NFL:15
Last Smatrix2:-0.00166729
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLR189.RCF with ID:NRCLR189
Raobcount :189
NFL:15
Last Smatrix2:-0.00167259
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKF189.RCF with ID:NRCKF189
Raobcount :196
NFL:15
Last Smatrix2:-0.00166637
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKI189.RCF with ID:NRCKI189
Raobcount :118
NFL:15
Last Smatrix2:-0.00167553
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCKQ189.RCF with ID:NRCKQ189
Raobcount :194
NFL:15
Last Smatrix2:-0.00170259
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLL189.RCF with ID:NRCLL189
Raobcount :188
NFL:15
Last Smatrix2:-0.00169922
Found RCF file:/home/local/raf/instruments/mtp/src/mtpbin/RCF/NRCLM189.RCF with ID:NRCLM189
Raobcount :192
NFL:15
Last Smatrix2:-0.00170313
